src/validation.cpp

```diff
@@ -5045,12 +5046,30 @@ bool LoadMempool(CTxMemPool& pool)
         for (const auto& i : mapDeltas) {
             pool.PrioritiseTransaction(i.first, i.second);
         }
+
+        // check if there are unbroadcast transactions
+        bool has_unbroadcast_txns = false;
+
+        try {
```

Comment:
this is the best option I've been able to identify for how to check if we are at the end of the file. would love to hear about any cleaner solutions.